### DISCUSSION TITLE
Deployment using GCP services

### DISCUSSION THREAD

**Discussion by MaxTeiger** [saleor/saleor#10719] [Deployment] [answered]
Hello ! 👋🏼 

I wanted to deploy Saleor on my GCP Project, so I started to run into the documentation. Using saleor-platform everything went fine and I could run the application in local without any problem.

This is the architecture I was targeting for after reading through your documentation : 
<img width="1580" alt="image" src="https://user-images.githubusercontent.com/38110296/191986777-f40e3ea6-1d90-48b0-b5a1-4b97ffc13e8c.png">

So, I build the image for Saleor based on version 3.7.11 --> Ok ✅ 
I build the dashboard using `npm i` `npm run build` using the (I think) right environments variables to generate the bundle --> Ok ✅ 

But now I don't really know what to do now. 

- If I check the file `build/dashboard/index.html` in a browser I only have a blank page even if saleor-core is running fine at http://localhost:8000 --> How am I supposed to configure Saleor Dashboard if I want to host it on a static hosting platform (here Google Cloud Storage)  ?
- If I run my saleor builded image outside the docker-compose (but using the same command to start the container), it doesn't work either --> what is the configuration needed so I can have my api working with my DB & Redis in GCP ? (and the dashboard deployed on GCS)

**Comment by peelar** [comment]
Hello @MaxTeiger 👋 

> If I check the file build/dashboard/index.html in a browser I only have a blank page even if [saleor-core](https://app.clickup.com/t/2549495/SALEOR-core) is running fine at http://localhost:8000/ --> How am I supposed to configure Saleor Dashboard if I want to host it on a static hosting platform (here Google Cloud Storage) ?

Did you have a look at the [Saleor Dashboard README file](https://github.com/saleor/saleor-dashboard#configuration)? If so, can you please share your env vars?

> If I run my saleor builded image outside the docker-compose (but using the same command to start the container), it doesn't work either --> what is the configuration needed so I can have my api working with my DB & Redis in GCP ? (and the dashboard deployed on GCS)

We will need more details here. Can you please provide a step-by-step description of what you did and what is not working?

**Reply by MaxTeiger** [reply]
Hello @peelar 👋🏼 

Yes, sorry for the incomplete information, I didn't have my logs under my eyes when I wrote the message. 

I deployed a debug version of my api at https://saleor-core.arianee.net

But I get the error message : `column channel_channel.allocation_strategy does not exist` when I run queries on it, despite running the migration smothly using the docker-compose in [saleor-platform](https://github.com/saleor/saleor-platform).

I got the same error when I tried to connect using the dashboard from my localhost. 

<img width="467" alt="image" src="https://user-images.githubusercontent.com/38110296/192584759-224e0ea2-a4e7-49c5-a1a7-6dbed958a3f5.png">


NB: 
FYI my env var for Saleor-Core are :

Name | Values 
-- | -- | 
ALLOWED_HOSTS | saleor-core.arianee.net 
DATABASE_URL | postgres://postgres:XXX@10.10.10.10:5432/postgres 
CELERY_BROKER_URL | redis://10.10.10.9:6379 
REDIS_URL | redis://10.10.10.9:6379 
SECRET_KEY | changeme 
GS_BUCKET_NAME | test-saleor-arianee 
GOOGLE_APPLICATION_CREDENTIALS | /secrets/saleor-sa-bucket-key
DEBUG | True

And I am running saleor-core `v3.7.11`

If you need more information, I will be happy to give it to you 😄

**Comment by MaxTeiger** [comment, marked as answer]
I found an answer to my problem, the script migrate database was run with the latest version of Saleor (my bad), but I was running the API using v3.7.11 so I think the DB mismatched. I just ran : 

```bash
git checkout tags/3.7.11
docker build . -t saleor
docker run --rm saleor python3 manage.py migrate
```

And the problem was solve. I am planning on writing an HowTo deploy with this architecture if you're interested !

Have a great day

**Reply by peelar** [reply]
I am glad you managed to solve it. We are always grateful to see content from our community members, let me know if you need any help writing it down :).

**Reply by herobaby71** [reply]
Hi @MaxTeiger, I am looking to deploy Saleor Core to GCP and need some of your advice since you have already set it up. I wonder if I should deploy everything on a single Compute Engine fronted by a Load Balancer or set up a Kubernetes cluster for the Saleor Core and host the front ends on Vercel. 

Do you have any recommendations before I go down this path since you have already set it up? Cheers!

**Comment by timuric** [comment]
[Answer removed to not confuse readers with outdated info]

**Reply by AliAlnajjar** [reply]
I have checked the prices [here](https://saleor.io/pricing). the cheapest for production use is the **Pro** account which costs **1,295** USD/mo !!
Am I missing something here ?

**Reply by timuric** [reply]
Sorry, the pricing has changed since then
